### ISSUE TITLE
pkgin module fails with package names with two hyphens on NetBSD 6.1.5

### ISSUE THREAD

**Issue by velotraveler** [ansible/ansible-modules-extras#2846]
##### ISSUE TYPE
- Bug Report
  ##### COMPONENT NAME
  - pkgin
##### ANSIBLE VERSION

```
ansible 2.0.2.0
  config file = 
  configured module search path = Default w/o overrides

```
##### CONFIGURATION

No use of ANSIBLE_\* environment vars, no use of ansible.cfg
##### OS / ENVIRONMENT

MacOS Yosemite 10.10.5
##### SUMMARY

pkgin module fails to parse the results of running NetBSD pkgin command when trying to install or check status of packages whose names have two hyphens.  In my environment, I can successfully install "mutt" or "bash-doc" packages using the pkgin module, but not "p5-NetAddr-IP" or "p5-HTML-Parser"
##### STEPS TO REPRODUCE

```
    - name: SpamAssasin Perl prerequisites
      pkgin: name={{ item }} state=present
      with_items:
        - p5-NetAddr-IP
        - p5-HTML-Parser

```

I also tried using syntax of multiple package names joined with commas, no difference.
##### EXPECTED RESULTS

Anything but a walkback :-).  For package names with one or fewer hyphens, modules get installed or reported on as no change needed as one would expect from module documentation.
##### ACTUAL RESULTS

walkback - "ValueError: need more than 1 value to unpack"

note: output has been slightly edited to anonymize hostname and username.

```
No config file found; using defaults
Loaded callback default of type stdout, v2.0

PLAYBOOK: site.yml *************************************************************
1 plays in site.yml

PLAY [mailinglist_servers] *****************************************************

TASK [setup] *******************************************************************
<myhost.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.example.com> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r myhost.example.com '/bin/sh -c '"'"'mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1472960163.17-63038818773270 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1472960163.17-63038818773270 `"'"'"''
<myhost.example.com> PUT /var/folders/wk/b7cvk27s4y55k480sw4s4xf40000gn/T/tmpFaB3NT TO /root/.ansible/tmp/ansible-tmp-1472960163.17-63038818773270/setup
<myhost.example.com> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r '[myhost.example.com]'
<myhost.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.example.com> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r -tt myhost.example.com '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/pkg/bin/python2.7 /root/.ansible/tmp/ansible-tmp-1472960163.17-63038818773270/setup; rm -rf "/root/.ansible/tmp/ansible-tmp-1472960163.17-63038818773270/" > /dev/null 2>&1'"'"''
ok: [myhost.example.com]

TASK [spamassassin : SpamAssasin Perl prerequisites] ***************************
task path: /Users/testuser/deploy/roles/spamassassin/tasks/main.yml:7
<myhost.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.example.com> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r myhost.example.com '/bin/sh -c '"'"'mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470 `"'"'"''
<myhost.example.com> PUT /var/folders/wk/b7cvk27s4y55k480sw4s4xf40000gn/T/tmpE3mOTB TO /root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/pkgin
<myhost.example.com> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r '[myhost.example.com]'
<myhost.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.example.com> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r -tt myhost.example.com '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/pkg/bin/python2.7 /root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/pkgin; rm -rf "/root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/" > /dev/null 2>&1'"'"''
failed: [myhost.example.com] (item=p5-NetAddr-IP) => {"failed": true, "invocation": {"module_name": "pkgin"}, "item": "p5-NetAddr-IP", "module_stderr": "OpenSSH_6.2p2, OSSLShim 0.9.8r 8 Dec 2011\ndebug1: Reading configuration data /Users/testuser/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh_config\r\ndebug1: /etc/ssh_config line 20: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2108\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to myhost.example.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/pkgin\", line 2213, in <module>\r\n    main()\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/pkgin\", line 223, in main\r\n    install_packages(module, pkgin_path, pkgs)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/pkgin\", line 191, in install_packages\r\n    if query_package(module, pkgin_path, package):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960164.96-264677393923470/pkgin\", line 107, in query_package\r\n    pkgname_with_version, raw_state = package.split(splitchar)[0:2]\r\nValueError: need more than 1 value to unpack\r\n", "msg": "MODULE FAILURE", "parsed": false}
<myhost.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.example.com> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r myhost.example.com '/bin/sh -c '"'"'mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461 `"'"'"''
<myhost.example.com> PUT /var/folders/wk/b7cvk27s4y55k480sw4s4xf40000gn/T/tmpTt_Q4Y TO /root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/pkgin
<myhost.example.com> SSH: EXEC sftp -b - -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r '[myhost.example.com]'
<myhost.example.com> ESTABLISH SSH CONNECTION FOR USER: root
<myhost.example.com> SSH: EXEC ssh -C -vvv -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 -o ControlPath=/Users/testuser/.ansible/cp/ansible-ssh-%h-%p-%r -tt myhost.example.com '/bin/sh -c '"'"'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/pkg/bin/python2.7 /root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/pkgin; rm -rf "/root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/" > /dev/null 2>&1'"'"''
failed: [myhost.example.com] (item=p5-HTML-Parser) => {"failed": true, "invocation": {"module_name": "pkgin"}, "item": "p5-HTML-Parser", "module_stderr": "OpenSSH_6.2p2, OSSLShim 0.9.8r 8 Dec 2011\ndebug1: Reading configuration data /Users/testuser/.ssh/config\r\ndebug1: Reading configuration data /etc/ssh_config\r\ndebug1: /etc/ssh_config line 20: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_forwards: request forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 2108\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to myhost.example.com closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/pkgin\", line 2213, in <module>\r\n    main()\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/pkgin\", line 223, in main\r\n    install_packages(module, pkgin_path, pkgs)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/pkgin\", line 191, in install_packages\r\n    if query_package(module, pkgin_path, package):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1472960166.72-81621073318461/pkgin\", line 107, in query_package\r\n    pkgname_with_version, raw_state = package.split(splitchar)[0:2]\r\nValueError: need more than 1 value to unpack\r\n", "msg": "MODULE FAILURE", "parsed": false}
...ignoring

PLAY RECAP *********************************************************************
myhost.example.com          : ok=2    changed=0    unreachable=0    failed=0   


```

**Comment by velotraveler** [comment]
to clarify, Ansible manager host is MacOS Yosemite, managed host is NetBSD 6.1.5.

**Comment by ansibot** [comment]
@L2G, @szinck, @troy2914, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by troy2914** [comment]
needs_info
I believe this was fixed in 2.1 8c6a3e7 on Apr 14
comment was  "pkgin: Fix bad regexp which did not catch packages such as p5-SVN-Notify"

which version of pkgin were you using?

**Comment by velotraveler** [comment]
I don't see any way to tell the version of an installed Ansible module, there's nothing in pkgin.py that looks like a local version number.

I installed ansible via MacPorts. I just tried upgrading, the old and new MacPorts versions are:
--->  Deactivating ansible @2.0.2.0.1_0
...
--->  Activating ansible @2.1.1.0.1_0

And the Ansible pkgin module now works as expected.

The patch referred to above was not in the 2.0.2.1_0 version, but it is in the 2.1.1.0.1_0.  I thought I had tried upgrading before opening this issue, perhaps I hadn't done "port selfupdate" first.

Thank you!
